external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaContinuousSinkSuite.scala

```diff
@@ -422,4 +349,54 @@ class KafkaContinuousSinkSuite extends KafkaContinuousTest {
     withOptions.foreach(opt => stream.option(opt._1, opt._2))
     stream.start()
   }
+
+  private def runAndVerifyStreamingQueryException(
+      inputTopic: String,
+      expectErrorMsg: String)(
+      writerFn: => StreamingQuery): Unit = {
+    var writer: StreamingQuery = null
+    val ex: Exception = try {
+      intercept[StreamingQueryException] {
+        writer = writerFn
+        testUtils.sendMessages(inputTopic, Array("1", "2", "3", "4", "5"))
+        eventually(timeout(streamingTimeout)) {
+          assert(writer.exception.isDefined)
+        }
+        throw writer.exception.get
+      }
+    } finally {
+      if (writer != null) writer.stop()
+    }
+    assert(ex.getCause.getCause.getMessage.toLowerCase(Locale.ROOT).contains(expectErrorMsg))
+  }
+
+  private def verifyException[T <: Exception : ClassTag](
+      expectErrorMsg: String)(
+      writerFn: => StreamingQuery): Unit = {
+    var writer: StreamingQuery = null
+    val ex: Exception = try {
+      intercept[T] {
+        writer = writerFn
+      }
+    } finally {
+      if (writer != null) writer.stop()
+    }
+    assert(ex.getMessage.toLowerCase(Locale.ROOT).contains(expectErrorMsg))
+  }
+
+  private def assertWrongSchema(
+      input: DataFrame,
+      selectExpr: Seq[String],
+      expectErrorMsg: String): Unit = {
+    verifyException[AnalysisException](expectErrorMsg)(
```

Comment:
Can you reuse `runAndVerifyStreamingQueryException` here like in the other suite? These seem to expect the error to happen when the writer is created, so the stuff that method does after it calls `writeFn` shouldn't influence the test result.

That could make it possible to reuse these for both test suites (e.g. by adding them to `KafkaTestUtils`).

Reply:
I separated both because one does need actual input topic and the other doesn't, but no harm to provide input topic for latter as well. I'll make a change.

Btw, there's some difference of waiting the query result and checking exception between batch/micro-batch and continuous so it doesn't seem to be a duplication. Actually the purpose of this PR was deduplicating the code which is due to the number of fields, and scope seems to be continuously increasing. Maybe we can revisit deduplicating code between batch/micro-batch and continuous once more in follow-up PR. WDYT?